source/MRViewer/ImGuiMenu.cpp

```diff
@@ -3408,17 +3408,22 @@ void ImGuiMenu::UiRenderManagerImpl::postRenderViewport( ViewportId viewport )
 
 BasicUiRenderTask::BackwardPassParams ImGuiMenu::UiRenderManagerImpl::beginBackwardPass()
 {
-    return { .mouseHoverConsumed = ImGui::GetIO().WantCaptureMouse };
+    return { .mouseHoverConsumed = !allowMouseInteraction() };
 }
 
 void ImGuiMenu::UiRenderManagerImpl::finishBackwardPass( const BasicUiRenderTask::BackwardPassParams& params )
 {
-    mouseIsBlocked_ = params.mouseHoverConsumed;
+    ownsMouseHover_ = params.mouseHoverConsumed && allowMouseInteraction();
 }
 
-bool ImGuiMenu::UiRenderManagerImpl::ownsMouseHover() const
+bool ImGuiMenu::UiRenderManagerImpl::shouldBlockMouseDown( int buttonIndex ) const
 {
-    return mouseIsBlocked_;
+    return buttonIndex == int( Viewer::MouseButton::Left ) && ownsMouseHover_;
```

Comment:
why check only left?

Reply:
Because we want middle/right buttons to work to move/rotate the camera even when a name tag is hovered.